packages/core/src/Page/index.js

```diff
@@ -590,16 +589,20 @@ class Page {
     // Set expressive layout file as an includedFile
     this.includedFiles.add(layoutPagePath);
     return fs.readFileAsync(layoutPagePath, 'utf8')
-      // Include file but with altered cwf (the layout page)
-      // Also render MAIN_CONTENT_BODY back to itself
+      /*
+       Render {{ MAIN_CONTENT_BODY }} and {% raw/endraw %} back to itself first,
+       which is then dealt with in the call below to {@link renderSiteVariables}.
+       */
+      .then(result => this.variableProcessor.renderPage(layoutPagePath, result, {
```

Comment:
variable rendering is moved out of `includeFile`, also in `page.generate() / resolveDependency()`